### PR TITLE
ticket 7 patch request on article_id

### PR DESCRIPTION
**Pull request by claire-farrow**
- patch - inc_votes with positive number - status 200
- patch - inc_votes with negative number - status 200
- patch - status 400 invalid inv_votes request

### CODE REVIEW

File: __tests__/app.test.js [hunk 1]
```diff
@@ -64,4 +64,48 @@ describe('/api/articles/', () => {
...
+                    created_at: expect.any(String),
+                    votes: -100
+                }))
+        })
+        
+    })
+
+    test('status: 400 - reponse with err msg for invalid inc_vote request', () => {
+        const invalidVote = {inc_votes: NaN}
+        return request(app).patch("/api/articles/3").send(invalidVote).expect(400).then(({ body: { msg } }) => {
```
Comment (popatre) [inline __tests__/app.test.js:106]: tests -  what if the key is spelt wrong?
 - what if there is an extra key on the patch object eg. title?

File: __tests__/app.test.js [hunk 2]
```diff
@@ -64,4 +64,48 @@ describe('/api/articles/', () => {
...
+            expect(msg).toBe("Bad Request");
+        })
```
Comment (popatre) [inline __tests__/app.test.js:108]: tests - patch to an article that doesnt exist?

**Review by popatre — CHANGES_REQUESTED** [review]

File: __tests__/app.test.js (hunk 1 above)
Comment (claire-farrow) [reply __tests__/app.test.js:104]: Added these two tests

File: __tests__/app.test.js (hunk 2 above)
Comment (claire-farrow) [reply __tests__/app.test.js:106]: added this test

File: models/topics.models.js [hunk 3]
```diff
@@ -8,12 +8,26 @@ exports.fetchTopics = () => {
...
             return Promise.reject({status: 404, msg: "Article Not Found"});
        }
-       return { article: rows[0] };
-       
+       return rows[0];
    })
+}
+
+exports.updateArticleById = (article_id, newVote) => {
+    const {inc_votes} = newVote
```
Comment (Augs0) [inline models/topics.models.js:21]: This approach seems a bit convoluted - why do we need to pop anything here?

File: __tests__/app.test.js [hunk 4]
```diff
@@ -64,4 +64,64 @@ describe('/api/articles/', () => {
...
+    })
+    test('status: 400 - reponse with err msg when extra key is added to the inc_vote request', () => {
+        const invalidVote = {inc_votes: 3, title: 'invalid'}
+        return request(app).patch("/api/articles/3").send(invalidVote).expect(400).then(({ body: { msg } }) => {
+            expect(msg).toBe("Bad Request");
+        })
+    })
+    test('Status: 404 - responds with error msg for VALID but Non-existent article_id', () => {
```
Comment (Augs0) [inline __tests__/app.test.js:120]: Only one 404 test needed - it shouldn't be tied to any one path

File: models/topics.models.js (hunk 3 above)
Comment (claire-farrow) [reply models/topics.models.js:21]: If I don't pop inc_votes the if statement to run the error code doesnt work

**Pull request merged**